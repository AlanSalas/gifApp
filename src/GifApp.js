import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifContainer from "./components/GifContainer";

const GifApp = () => {
  const [categories, setCategories] = useState(["Welcome"]);

  return (
    <div className="gif-app">
      <h2 className="gif-app__title">Gif App</h2>
      <AddCategory setCategories={setCategories} />

      {categories.map((category) => (
        <GifContainer key={category} category={category} />
      ))}
    </div>
  );
};

export default GifApp;
