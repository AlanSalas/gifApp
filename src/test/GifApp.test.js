import { shallow } from "enzyme";
import GifApp from "../GifApp";

describe("Tests in GifApp component", () => {
  const wrapper = shallow(<GifApp />);

  test("Should render propertly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should have at least 1 GifContainer", () => {
    expect(wrapper.find("GifContainer").length).toBe(1);
  });
});
