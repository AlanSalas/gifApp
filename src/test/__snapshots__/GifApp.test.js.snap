// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifApp component Should render propertly 1`] = `
<div
  className="gif-app"
>
  <h2
    className="gif-app__title"
  >
    Gif App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <GifContainer
    category="Welcome"
    key="Welcome"
  />
</div>
`;
